diff --git a/ChangeLog b/ChangeLog
index dbddbc0..ead5da5 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -2,6 +2,10 @@ $Id: ChangeLog 5155 2020-12-30 16:49:18Z chrfranke $
 
 2021-01-17  Christian Franke  <franke@computer.org>
 
+	drivedb.h:
+	- Bump branch VERSION to 7.3.
+	- USB: ASMedia ASM2362 (0x174c:0x2362).
+
 	scsinvme.cpp: Add '-d sntasmedia' device type for ASMedia ASM2362
 	USB to NVMe bridges (#1221).
 	dev_interface.cpp: Update help text.
diff --git a/drivedb.h b/drivedb.h
index 400cc29..0253f78 100644
--- a/drivedb.h
+++ b/drivedb.h
@@ -5858,6 +5858,12 @@ const drive_settings builtin_knowndrives[] = {
     "-d sat"
   },
   // ASMedia
+  { "USB: ; ASMedia ASM2362", // USB->PCIe (NVMe)
+    "0x174c:0x2362",
+    "",
+    "",
+    "-d sntasmedia"
+  },
   { "USB: ; ASMedia",
     "0x174c:0x....",
     "",
