From d5d7c4078460597358aaf635c431c53705081326 Mon Sep 17 00:00:00 2001
From: Tesla OpenSource <opensource@tesla.com>
Date: Sat, 18 Apr 2020 15:00:50 -0700
Subject: [PATCH] Remove redundant gettext file

---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e4c6aae..3293833 100644
--- a/configure.ac
+++ b/configure.ac
@@ -690,7 +690,6 @@ AM_CONDITIONAL(USE_NLS, test "x$USE_NLS" = "xyes")
 
 AC_CONFIG_FILES([
 	Makefile
-	po/Makefile.in
 	doc/Makefile
 	man/Makefile
 	man/config.xml
-- 
2.19.2

