--- busybox-1.24.2/init/bootchartd.c.orig	2017-01-05 16:58:34.616680868 -0800
+++ busybox-1.24.2/init/bootchartd.c	2017-01-05 17:41:51.812000335 -0800
@@ -79,6 +79,11 @@
 //# Sampling period (in seconds)
 //SAMPLE_PERIOD=0.2
 //
+//# Whether to automatically stop logging as the boot process completes.
+//# The logger will look for known processes that indicate bootup completion
+//# at a specific runlevel (e.g. gdm-binary, mingetty, etc.).
+//AUTO_STOP_LOGGER="yes"
+//
 //not yet supported:
 //# tmpfs size
 //# (32 MB should suffice for ~20 minutes worth of log data, but YMMV)
@@ -93,11 +98,6 @@
 //# Tarball for the various boot log files
 //BOOTLOG_DEST=/var/log/bootchart.tgz
 //
-//# Whether to automatically stop logging as the boot process completes.
-//# The logger will look for known processes that indicate bootup completion
-//# at a specific runlevel (e.g. gdm-binary, mingetty, etc.).
-//AUTO_STOP_LOGGER="yes"
-//
 //# Whether to automatically generate the boot chart once the boot logger
 //# completes.  The boot chart will be generated in $AUTO_RENDER_DIR.
 //# Note that the bootchart package must be installed.
@@ -214,13 +214,13 @@ static char *make_tempdir(void)
 	return tempdir;
 }
 
-static void do_logging(unsigned sample_period_us, int process_accounting)
+static void do_logging(unsigned sample_period_us, int process_accounting, int auto_stop_logger)
 {
 	FILE *proc_stat = xfopen("proc_stat.log", "w");
 	FILE *proc_diskstats = xfopen("proc_diskstats.log", "w");
 	//FILE *proc_netdev = xfopen("proc_netdev.log", "w");
 	FILE *proc_ps = xfopen("proc_ps.log", "w");
-	int look_for_login_process = (getppid() == 1);
+	int look_for_login_process = auto_stop_logger && (getppid() == 1);
 	unsigned count = 60*1000*1000 / sample_period_us; /* ~1 minute */
 
 	if (process_accounting) {
@@ -347,6 +347,7 @@ int bootchartd_main(int argc UNUSED_PARA
 	pid_t parent_pid, logger_pid;
 	smallint cmd;
 	int process_accounting;
+	int auto_stop_logger;
 	enum {
 		CMD_STOP = 0,
 		CMD_START,
@@ -381,6 +382,7 @@ int bootchartd_main(int argc UNUSED_PARA
 	/* Read config file: */
 	sample_period_us = 200 * 1000;
 	process_accounting = 0;
+	auto_stop_logger = 1;
 	if (ENABLE_FEATURE_BOOTCHARTD_CONFIG_FILE) {
 		char* token[2];
 		parser_t *parser = config_open2("/etc/bootchartd.conf" + 5, fopen_for_read);
@@ -394,6 +396,10 @@ int bootchartd_main(int argc UNUSED_PARA
 			) {
 				process_accounting = 1;
 			}
+			if (strcmp(token[0], "AUTO_STOP_LOGGER") == 0 && token[1] &&
+		            (strcmp(token[1], "off") == 0 || strcmp(token[1], "no") == 0)) {
+				auto_stop_logger = 0;
+			}
 		}
 		config_close(parser);
 		if ((int)sample_period_us <= 0)
@@ -426,7 +432,7 @@ int bootchartd_main(int argc UNUSED_PARA
 			putenv((char*)bb_PATH_root_path);
 
 		tempdir = make_tempdir();
-		do_logging(sample_period_us, process_accounting);
+		do_logging(sample_period_us, process_accounting, auto_stop_logger);
 		finalize(tempdir, cmd == CMD_START ? argv[2] : NULL, process_accounting);
 		return EXIT_SUCCESS;
 	}
